### DISCUSSION TITLE
Problem with ~1GB file in installer

### DISCUSSION THREAD

**Discussion by dmagee** [wixtoolset/issues#8035] [Questions]
I have an installer that's working fine until I add a particularly large file (~1GB). I put it in it's own cab file within the .msi. WiX builds the installer ok, but then I get the error "The system cannot open the device or file selected" and then "The installer has encountered an unexpected error installing this package. This may indicate a problem with the package. The error code is 2755.". If I remove this large file all works fine. I'm not sure what to do. cab files are supposed to be ok up to 2GB? This file is 1,021,297,704 bytes. Thanks in advance!

**Comment by ds-andreasor** [comment]
This is missing some more info about your authoring.
I have msi packages (x64) with external cabs that are above the 2GB size and they work fine in v3 and 4.0.4.

I only remember that v3 burn bundle (x86) could not handle embedding large msi packages if the total exceeds 2 GB in size.
If these large msis are external payloads it is fine.

**Comment by dmagee** [comment]
I've also made plenty of big installers, but this is possibly the biggest single file I've tried to embed. I've been using Media tags as below:

```
      <Media Id="1" Cabinet="media1.cab" EmbedCab="yes" />
      <Media Id="2" Cabinet="media2.cab" EmbedCab="yes" />
      <Media Id="3" Cabinet="media3.cab" EmbedCab="yes" />
...
<Component Id="COMPEXENM_43" Guid="e6ba7fa8-bac3-4bb2-9dfc-9fab71a0549f" Win64="$(var.Win64)">
    <File Source="C:\..\my_big_1GBfile.dll" Id="FILEEXENM_43" KeyPath="yes" DiskId="3" />
</Component>
...
      <!-- Features define which parts of the application can be installed in a custom installation -->
      <Feature Id="Complete" Title="!(loc.ApplicationName)" Description="!(loc.FeatureCompleteDescription)" Display="expand" Level="1" ConfigurableDirectory="INSTALLDIR">

         <!-- A feature block for the main (GUI) program and all its dependencies -->
         <Feature Id="MainProgram" Title="!(loc.FeatureMainProgramTitle)" Description="!(loc.FeatureMainProgramDescription)" Level="1">
            <ComponentRef Id="INSTALLDIR_comp" />
...
                 <ComponentRef Id="COMPEXENM_43" />
....			
           <!-- Registry entries -->
            <ComponentRef Id="RegValInstallLocation_comp" />
         </Feature>

      </Feature>

```

All components are within a single feature, and the big file is the only thing in media3.cab. I've just been reading about MediaTemplate with MaximumCabinetSizeForMediaSplitting, so I may try that next. Do I need multiple features too? I have never had to do that for large msi files before. The total msi is about 2.3GB, which isn't particularly big compared to others I've created.

**Reply by dmagee** [reply]
Solved it! using  MediaTemplate with MaximumCabinetSizeForMediaSplitting='1024', rather than manually assigning files to multiple Media tags seems to have fixed it!
